Sming/Makefile

```diff
@@ -267,12 +267,13 @@ endif
 MFORCE32 := $(shell $(CC) --help=target | grep mforce-l32)
 
 # compiler flags using during compilation of source files. Add '-pg' for debugging
-CFLAGS += -Wpointer-arith -Wundef -Werror -Wl,-EL -nostdlib -mlongcalls -mtext-section-literals -finline-functions -fdata-sections -ffunction-sections \
-         -D__ets__ -DICACHE_FLASH -DARDUINO=106 -DCOM_SPEED_SERIAL=$(COM_SPEED_SERIAL) -DENABLE_CMD_EXECUTOR=$(ENABLE_CMD_EXECUTOR) -DESP8266=1 -DSMING_INCLUDED=1 
+CFLAGS += -Wall -Wno-comment \
```

Comment:
> I may be a little premature with this one but it does highlight the areas I think need to be sorted before 3.7.x. I have PRs to fix most of the remaining problems

Can you add a directive that switches to the higher level of error reporting? For example STRICT=1. Strict by default should be disabled. What this will bring is:
* not scare new, returning or non-core Sming/C++ developers
* but at the same time allow us, the core developers, to see the issues and take care to gradually fix them (the travis and appveyor build should have STRICT=1 by default).

Once all major and minor issues are resolved we can set the STRICT directive to be `on` by default.

Reply:
Good idea. Thinking about `-Werror`, I'll put that back in for non-STRICT but perhaps leave it out for `STRICT` might be more helpful for development. What do you think?

Reply:
I've also deliberately left `Makefile-rboot.mk` and `Makefile-project.mk` out of all  this - something for future consideration perhaps...